hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/safemode/SafeModeHandler.java

```diff
@@ -129,7 +129,8 @@ private void cleanupPipelines() {
     List<Pipeline> pipelineList = scmPipelineManager.getPipelines();
     pipelineList.forEach((pipeline) -> {
       try {
-        if (pipeline.getPipelineState() == Pipeline.PipelineState.ALLOCATED) {
+        if (pipeline.getPipelineState() == Pipeline.PipelineState.ALLOCATED &&
+            pipeline.isAllocationTimeout()) {
           scmPipelineManager.finalizeAndDestroyPipeline(pipeline, false);
```

Comment:
This condition is currently always false because pipeline.isAllocationTimeout() always return false.

Reply:
Right, there is a TODO.  Will complete the logic in HDDS-2177, "Add a srubber thread to detect creation failure pipelines in ALLOCATED state".  
This patch is too big. I'd like to keep mandatory change in this patch and have others in new tasks.